cmd/server/config.go

```diff
@@ -59,25 +58,21 @@ func (c *config) ApplyDefaults() {
 		c.IngestionMode = service.StandardIngestion
 	}
 
-	if c.RPCEndpoint == "" {
-		c.RPCEndpoint = "http://localhost:9650"
+	if c.RPCBaseURL == "" {
+		c.RPCBaseURL = "http://localhost:9650"
 	}
 
-	if c.IndexerEndpoint == "" {
-		c.IndexerEndpoint = c.RPCEndpoint
+	if c.IndexerBaseURL == "" {
+		c.IndexerBaseURL = c.RPCBaseURL
 	}
 
 	if c.ListenAddr == "" {
 		c.ListenAddr = "0.0.0.0:8080"
 	}
 }
 
-func (c *config) Validate() error {
-	c.ApplyDefaults()
-
-	if c.RPCEndpoint == "" {
-		return errMissingRPC
-	}
+func (c *config) validate() error {
+	c.applyDefaults()
```

Comment:
nit:  you can probably call `applyDefaults` between `readConfig` and `validate` to logically separate the steps

Reply:
Done